database/etcddb/storage.go

```diff
@@ -0,0 +1,205 @@
+package etcddb
+
+import (
+	"github.com/aptly-dev/aptly/database"
+	"github.com/pborman/uuid"
+	clientv3 "go.etcd.io/etcd/client/v3"
+
+	"fmt"
+)
+
+type EtcDStorage struct {
+	url       string
+	db        *clientv3.Client
+	tmpPrefix string // prefix for temporary DBs
+}
+
+// CreateTemporary creates new DB of the same type in temp dir
+func (s *EtcDStorage) CreateTemporary() (database.Storage, error) {
+	tmp := uuid.NewRandom().String()
+	return &EtcDStorage{
+		url:       s.url,
+		db:        s.db,
+		tmpPrefix: tmp,
+	}, nil
+}
+
+func (s *EtcDStorage) applyPrefix(key []byte) []byte {
+	if len(s.tmpPrefix) != 0 {
+		return append([]byte(s.tmpPrefix+"/"), key...)
+	}
+	return key
+}
+
+// Get key value from etcd
+func (s *EtcDStorage) Get(key []byte) (value []byte, err error) {
+	key = s.applyPrefix(key)
```

Comment:
Wouldn't be clearer to use another name here ? Like `prefixedKey` ?

Reply:
maybe, however, the prefix is only added for temdbs. maybe applyPrefix should be maybePrefix ?

Reply:
Quite not a fan of `maybe` vars, it let us think that the behaviour might not be the one intended.

What do you think of `wantedKey` or `realKey` ?